src/Accounting/index.ts

```diff
@@ -344,3 +345,55 @@ export const calcUnrealised: (
     let avgPrice = sumOfAmounts.div(sumOfWeights);
     return (base.times(price)).minus(base.times(avgPrice));
 }
+
+/**
+ * Given a users position and a trade, returns the users new position after the trade has been applied
+ * @param position the users current position
+ * @param trade the trade to be performed
+ * @param feeRate fee rate of the market eg. 0.02 for 2%
+ * @returns the users new position
+ */
+export const calcPositionAfterTrade: (
+    position: Position,
+    trade: FlatOrderWithSide,
+    feeRate: BigNumber
+) => Position = (position, trade, feeRate) => {
+    const quoteChange = trade.amount.times(trade.price);
+    const fee = calcFee(trade.amount, trade.price, feeRate);
+
+    let newQuote = new BigNumber(0);
+    let newBase = new BigNumber(0);
+
+    // long
+    if(!trade.position) {
+        newBase = position.base.plus(trade.amount);
+        newQuote = position.quote.minus(quoteChange.plus(fee));
```

Comment:
Just noting. This is an open issue in Code Arena, so we may have to change this implementation

Reply:
See https://github.com/code-423n4/2021-06-tracer-findings/issues/127

Reply:
I think the best fix for the Code Arena issue is to implement the recommended change. I will update this to use the same fix (using a `-` instead of using parentheses to force the intended order of operations)